src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2Solubility.cpp

```diff
@@ -390,28 +234,18 @@ CO2Solubility::CO2Solubility( string const & name,
   string const expectedWaterPhaseNames[] = { "Water", "water", "Liquid", "liquid" };
   m_phaseLiquidIndex = PVTFunctionHelpers::findName( phaseNames, expectedWaterPhaseNames, "phaseNames" );
 
-  string const solubilityModel = 10 < inputParams.size() ? inputParams[10] : "DuanSun";
-  FunctionManager & functionManager = FunctionManager::getInstance();
-
-  if( solubilityModel == "SpycherPruess" )
-  {
-    std::pair< TableFunction const *, TableFunction const * > tables =
-      CO2SolubilitySpycherPruess::makeSolubilityTables( inputParams, m_modelName, functionManager );
-    m_CO2SolubilityTable = tables.first;
-    m_WaterVapourisationTable = tables.second;
-  }
-  else if( solubilityModel == "DuanSun" )
+  SolubilityModel solubilityModel = SolubilityModel::DuanSun;   // Default solubility model
+  if( inputParams[1] == EnumStrings< SolubilityModel >::toString( SolubilityModel::Tables ) )
```

Comment:
Not sure but wouldn't `EnumStrings< SolubilityModel >::fromString(s)` throw if pass any other values than the one defined ? If so, it will spare you the if statement and the silent defaulting